raft/src/test/java/org/apache/kafka/raft/RaftEventSimulationTest.java

```diff
@@ -550,14 +576,17 @@ void canAddObserverAndRemovePartitionedVoter(
 
         initializeClusterAndStartAppending(cluster, router, scheduler, 10);
 
-        // Partition a random voter, and add an observer to replace it
-        RaftNode voterToRemove = cluster.running.get(random.nextInt(numVoters));
-        router.filter(voterToRemove.id(), new DropAllTraffic());
+        // Partition a random voter. Add first observer as new voter
+        int voterIdToRemove = cluster.running.get(random.nextInt(numVoters)).id();
+        router.filter(voterIdToRemove, new DropAllTraffic());
+        expectedVoterIds.add(numVoters);
```

Comment:
nit: would be more readable if you used a new var like `firstsObserverId` and assigned to value of numVoters at beginning of test